lib/junk_drawer/rails/bulk_updatable.rb

```diff
@@ -11,14 +11,30 @@ def bulk_update(objects)
       objects = objects.select(&:changed?)
       return unless objects.any?
 
-      changed_attributes = extract_changed_attributes(objects)
-      query = build_query_for(objects, changed_attributes)
+      unique_objects = uniquify_and_merge(objects)
+      changed_attributes = extract_changed_attributes(unique_objects)
+      query = build_query_for(unique_objects, changed_attributes)
       connection.execute(query)
       objects.each(&:clear_changes_information)
     end
 
   private
 
+    def uniquify_and_merge(objects)
+      grouped_objects = objects.group_by(&:id).values
+      grouped_objects.each do |duplicates|
+        attrs = duplicates.each_with_object({}) do |duplicate, changes|
+          duplicate.changed.each do |changed_attribute|
+            changes[changed_attribute] = duplicate[changed_attribute]
```

Comment:
What if two instances each change the same attribute? I imagine we'd expect to keep whichever was updated later, or was ordered later in the set passed to `BulkUpdateble`, although I'm not sure to what extent that ordering is guaranteed. In any event, if it's possible to make this behavior consistent, should we make it explicit (and add a test for it)?

Reply:
Added an explicit test showing that the last assignment wins. This is basically the way it would perform if one were to not use `BulkUpdatable` and `save` records in place instead.